protostar-rust/src/main.rs

```diff
@@ -66,16 +77,38 @@ fn dependencies_for_package(
 fn main_execution() -> Result<()> {
     let _args = Args::parse();
 
+    let project_root_path = project_root::get_project_root()?
+        .to_str()
+        .expect("error reading project root path")
+        .to_owned();
     // TODO #1997
-    set_var("CARGO_MANIFEST_DIR", "../../cairo/Cargo.toml");
+    set_var(
+        "CARGO_MANIFEST_DIR",
+        format!("{}/../cairo/Cargo.toml", project_root_path),
+    );
 
     let scarb_metadata = MetadataCommand::new().inherit_stderr().exec()?;
 
+    let mut protostar_test_config = ProtostarTestConfig {
+        contracts_paths: HashMap::new(),
+    };
+    for package in &scarb_metadata.packages {
+        for target in &package.targets {
+            if target.kind == "starknet-contract" {
+                // TODO consider multiple targets of this kind
+                protostar_test_config
+                    .contracts_paths
+                    .insert(target.name.clone(), target.source_path.to_string());
+            }
+        }
```

Comment:
Why limit the runner to starknet-contract target? `dependencies_for_package` is defined in such way that it will fetch `starknet-contract` target dependencies by default, but if it's not present, it will use `lib` target as fallback, because it's always present in scarb.

I don't see any reason to only run on starknet target.

Reply:
Actually I think this approach is correct if we want to commit building contracts with scarb (I think that is the plan in the starknet CLI for now)
cc @MaksymilianDemitraszek

Reply:
We have to make this code working with what we have now. Once building is moved to rust, changes like this can be introduced.

Reply:
And if we already want to force the runner to only work with scarb packages with `starknet-contract` target, `dependencies_for_package` should be updated to only work with `starknet-contract` target.

I.e. this code should be updated
```rust
.min_by_key(|unit| match unit.target.name.as_str() {
            name @ "starknet-contract" => (0, name),
            name @ "lib" => (1, name),
            name => (2, name),
        })
        .ok_or_else(|| anyhow!("Failed to find metadata for package = {package}"))?;
```

Reply:
We can stick with MVP I guess so I would lean towards your first suggestion

Reply:
this is removed with building in scarb
[adapt for building in scarb](https://github.com/software-mansion/protostar/pull/2059/commits/b104f483c4ffb234aba0256cd0a8d988e727484b)